core/types/block.go

```diff
@@ -535,10 +535,10 @@ type RawBody struct {
 }
 
 type BodyForStorage struct {
-	BaseTxId uint64
-	TxAmount uint32
-	Uncles   []*Header
-	// TODO(yperbasis): withdrawals
+	BaseTxId    uint64
+	TxAmount    uint32
+	Uncles      []*Header
+	Withdrawals *[]*Withdrawal `rlp:"optional,nil"`
```

Comment:
Will the rlp optional logic work if the type is changed to `[]*Withdrawal` instead of `*[]*Withdrawal`?

Reply:
Unfortunately we lose the functionality of nil if we don't use a pointer type

Reply:
For normal slices in go there's a difference between a nil slice and and empty slice – you don't need a pointer for that.